homeassistant/components/shelly/config_flow.py

```diff
@@ -460,7 +472,7 @@ def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         return (
             get_device_entry_gen(config_entry) in RPC_GENERATIONS
             and not config_entry.data.get(CONF_SLEEP_PERIOD)
-            and config_entry.data.get("model") != MODEL_WALL_DISPLAY
+            and cast(bool, config_entry.data.get(CONF_SCRIPT))
```

Comment:
```suggestion
            and cast(bool, config_entry.data.get(CONF_SCRIPT, True))
```
So that scripts are not disabled for existing devices (all except for wall display already support scripts but they are not stored in entry data).

Reply:
We can then remove the check on Wall Display as it's redundant now

Reply:
> We can then remove the check on Wall Display as it's redundant now

But for wall display you want it to be disabled if the key is missing and for other devices to be enabled, you can't ignore the wall display

Reply:
OH I noticed now that I updated only 1 place and not both. Thx